### PR TITLE
[release/5.0-preview8] Hardcode path to ICU on macOS

### PR DESCRIPTION
**Pull request by ViktorHofer**
Port of https://github.com/dotnet/runtime/commit/05f17e3170d014efe63202c4e7e7b641e8f8fef6

### CODE REVIEW

**Comment by ViktorHofer** [comment]
As just discussed in Tactics for broken builds we don't need to wait for approval, hence merging.

**Review by akoeplinger — APPROVED** [review]

**Pull request merged**
